browser/BUILD.gn

```diff
@@ -170,6 +169,12 @@ source_set("browser_process") {
     ]
   }
 
+  if (!is_ios) {
+    deps += [
+      "//brave/components/brave_sync",
+    ]
+  }
```

Comment:
iOS doesn't actually need any of these… We'd be better to find the reason why `browser/BUILD.gn` is being included at all when building iOS

Reply:
I understand `components/brave_sync` must not be included into iOS build.

Here what I tried to check.

1) on current master I had removed `if (!is_ios) {` guard around
https://github.com/brave/brave-core/pull/4077/files#diff-91f51b969eaddd52f16043d734a68bd5L77
 `components/brave_sync/BUILD.gn` 
```
if (!is_ios) {	
    deps += [ "//content/public/browser" ]	
}
```

2) ensured iOS build fails with that change:
```
ERROR at //services/media_session/public/mojom/BUILD.gn:27:25: Assignment had no effect.
  export_header_blink = “third_party/blink/public/platform/web_common.h”
                        ^-----------------------------------------------
You set the variable “export_header_blink” here and it was unused before it went
out of scope.
See //content/public/browser/BUILD.gn:408:5: which caused the file to be included.
    “//services/media_session/public/mojom”,
    ^--------------------------------------
```

Note this is an error output from `gn gen`

3) put the guard at pt1 back to make possible run `gn path`

4) run `gn path out/ios_Release //brave //brave/components/brave_sync:prefs  --all  --with-data` and see
```
No paths found between these two targets.
```

Do similar check for `gn path out/ios_Release //brave //chrome/browser:browser  --all  --with-data` to ensure `chromium_src/chrome/browser/prefs/browser_prefs.cc` is not included into iOS build. Result also is 
```
No paths found between these two targets.
```

This makes me think `gn` does processing of all targets in sources dir. And it makes possible to see error if wrong deps as her in pt2.
It seems reasonable because the command `gn gen` does not know which target we are going to build with next `ninja` call.

So I am going just to revert guard at https://github.com/brave/brave-core/pull/4077/files#diff-84f8700538180eec90bb2f4c9dff4da4R172 .

Reply:
this should have a `enable_brave_sync` guard

Reply:
@kylehickinson it shouldn't cause problems for you after that because `enable_brave_sync` is tied to `enable_extensions` which is not enabled on ios or android

Reply:
pushed a fix with `enable_brave_sync` guard